File: 프로그래머스_고득점kit/Hash/폰켓몬.py
# n/2개를 가져도 됨
# [3,1,2,3] 종류가 있다면, 이 중 2개를 고를 때
# 최대한 많은 종류 선택 : 2종류(1,3), (1,2), (2,3)
def solution(nums):
    return min(len(nums)//2, len(set(nums)))

File: 프로그래머스_고득점kit/Hash/폰켓몬.java
import java.util.HashSet;

// 중복 제거 후 종류 수, nums//2 중 min 고르기
class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        int limit = nums.length / 2;
        // 중복제거 set 만들기
        HashSet<Integer> numSet = new HashSet<>();
        for (int num : nums) {
            numSet.add(num);
        }
        for (int num : numSet) {
            System.out.println(num);
        }
        
        answer = Math.min(numSet.size(), limit);
        return answer;
    }
}